docs/Style.thy

```diff
@@ -0,0 +1,182 @@
+(*
+ * Copyright 2021, Data61, CSIRO
+ *
+ * This software may be distributed and modified according to the terms of
+ * the BSD 2-Clause license. Note that NO WARRANTY is provided.
+ * See "LICENSE_BSD2.txt" for details.
+ *
+ * @TAG(DATA61_BSD)
+ *)
+
+chapter \<open>An Isabelle Syntax Style Guide\<close>
+
+theory Style
+  imports Main
+begin
+
+(* This style guide is heavily influenced by Gerwin's Style Guide for Isabelle/HOL. See parts 1
+   and 2 here:
+
+   * https://proofcraft.org/blog/isabelle-style.html
+   * https://proofcraft.org/blog/isabelle-style-part2.html
+
+   This guide may repeat some parts of Gerwin's guide and should not conflict in any way.
+ *)
+
+section \<open>General Principles\<close>
+
+(* TO DO: What advice should go here? *)
+
+
+section \<open>Definitions\<close>
+
+(* Definitions should be written over 2 lines in the following style.
+
+   Note that indentation is 2 spaces and the definitions do not run past 100 character line
+   limit. Note also that the \equiv symbol is used.
```

Comment:
\equiv is indeed our default. I wonder if we should make it mandatory or leave it open, because upstream style is to prefer the opposite (`=`). I do like \equiv better, mostly for reasons of syntax precedence (needs fewer parentheses). Upstream uses `<->` to get around the precedence for boolean definitions, but it doesn't work for other types and I find it strange for a definition.

The reason some people prefer `=` is that it's easier to work with in the logic, e.g. you can combine it easily with `sym` or similar without converting to object equality first, but I have never found that a hindrance in practice. Certainly not so far that I would write more parentheses everywhere.

Happy to keep it at \equiv, just wanted to point out that there are other opinions.